source/v5/mqtt3_to_mqtt5_adapter.c

```diff
@@ -1663,11 +1678,7 @@ static uint16_t s_aws_mqtt_client_connection_5_publish(
 
     uint16_t synthetic_id = operation->base.id;
 
-    /* transient ref on adapter while moving to the event loop thread */
-    aws_mqtt3_to_mqtt5_adapter_operation_reference_adapter(&operation->base);
-
-    /* transient ref on operation while moving to the event loop thread */
-    aws_mqtt3_to_mqtt5_adapter_operation_acquire(&operation->base);
+    s_aws_mqtt3_to_mqtt5_adapter_operation_acquire_cross_thread_refs(&operation->base);
```

Comment:
Absolute nitpicking but should we leave the comment about why this is happening? It seemed useful.